src/librustdoc/html/render/mod.rs

```diff
@@ -4268,23 +4261,40 @@ fn get_methods(
 }
 
 // The point is to url encode any potential character from a type with genericity.
-fn small_url_encode(s: &str) -> String {
-    s.replace("<", "%3C")
-        .replace(">", "%3E")
-        .replace(" ", "%20")
-        .replace("?", "%3F")
-        .replace("'", "%27")
-        .replace("&", "%26")
-        .replace(",", "%2C")
-        .replace(":", "%3A")
-        .replace(";", "%3B")
-        .replace("[", "%5B")
-        .replace("]", "%5D")
-        .replace("\"", "%22")
+fn small_url_encode(s: String) -> String {
+    let mut st = String::new();
+    let mut last_match = 0;
+    for (idx, c) in s.bytes().enumerate() {
```

Comment:
I'm not sure if it's a good idea. We got bitten because of that recently (on characters which are more than one byte long). Please use [char_indices](https://doc.rust-lang.org/nightly/std/primitive.str.html#method.char_indices) instead.

Reply:
This is exactly what `Escape` does. I don't think there are continuation bytes that can have the same value as single byte characters, although I'm not terribly familiar with how UTF8 works.

Reply:
Well, in the end it'll be mostly the same code. Please update? :)

Reply:
Sure :)